### PR TITLE
[docs] best-practices add use variable with template example.

### PR DESCRIPTION
**Pull request by BobDu**
Signed-off-by: BobDu <i@bobdu.cc>

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow/best-practices.rst [hunk 1]
```diff
@@ -213,6 +213,30 @@ or if you need to deserialize a json object from the variable :
 
     {{ var.json.<variable_name> }}
 
+Ensure use variable with template in operator, not get it in top level code.
```
Comment (potiuk) [inline docs/apache-airflow/best-practices.rst]:
```suggestion
Make sure to use variable with template in operator, not in the top level code.
```

File: docs/apache-airflow/best-practices.rst [hunk 2]
```diff
@@ -213,6 +213,30 @@ or if you need to deserialize a json object from the variable :
...
+
+Bad example:
+
+.. code-block:: python
+
+    from airflow.models import Variable
+
+    foo_var = Variable.get("foo")
```
Comment (potiuk) [inline docs/apache-airflow/best-practices.rst]:
```suggestion
    foo_var = Variable.get("foo")  # DON'T DO THAT
```

File: docs/apache-airflow/best-practices.rst [hunk 3]
```diff
@@ -213,6 +213,30 @@ or if you need to deserialize a json object from the variable :
...
+
+.. code-block:: python
+
+    bash_use_variable_good = BashOperator(
+        task_id="bash_use_variable_good",
+        bash_command="echo variable foo=${foo_env}",
+        env={"foo_env": "{{ var.value.get('foo') }}"},
+    )
+
+
```
Comment (potiuk) [inline docs/apache-airflow/best-practices.rst:257]: Would be great to add also a bad example where variable is also (wrongly) used as parameter:
```
bash_command="echo variable foo=${Variable.get('foo')}"  # DON'T DO THAT
env={"foo_env": Variable.get('foo')}, # DON'T DO THAT
```

And another example, where we show how you CAN use Variable.get in @task - decorated code.

```
@task
def my_task():
     var = Variable.get('foo')   # this is fine
     print(var)

**Review by potiuk — CHANGES_REQUESTED** [review]
Good Idea.

However, I think a few additional examples and making it clear what's good/not good will make it even better.

File: docs/apache-airflow/best-practices.rst (hunk 3 above)
Comment (BobDu) [reply docs/apache-airflow/best-practices.rst:257]: ? `bash_command="echo variable foo=${Variable.get('foo')}"` is not a effective syntax.
Are you want to say `bash_command=f"echo variable foo={Variable.get('foo')}"` ?

**Review by potiuk — APPROVED** [review]
Love it.

**Pull request merged**